rubicon/objc/types.py

```diff
@@ -54,16 +232,25 @@ def encoding_for_ctype(vartype):
     NSRangeEncoding = b'{_NSRange=II}'
     UIEdgeInsetsEncoding = b'{UIEdgeInsets=ffff}'
     NSEdgeInsetsEncoding = b'{NSEdgeInsets=ffff}'
+    PyObjectEncoding = b'^{_object=i^{_typeobject}}'
 
-NSIntegerEncoding = encoding_for_ctype(NSInteger)
-NSUIntegerEncoding = encoding_for_ctype(NSUInteger)
-CGFloatEncoding = encoding_for_ctype(CGFloat)
+NSIntegerEncoding = ctype_to_encoding(NSInteger)
+NSUIntegerEncoding = ctype_to_encoding(NSUInteger)
+CGFloatEncoding = ctype_to_encoding(CGFloat)
 
 # Special case so that NSImage.initWithCGImage_size_() will work.
 CGImageEncoding = b'{CGImage=}'
 
 NSZoneEncoding = b'{_NSZone=}'
 
+register_preferred_encoding(PyObjectEncoding, py_object)
+
+
+class UnknownPointer(c_void_p):
+    """Placeholder for the b'^?' "unknown pointer" type. Not to be confused with a b'^v' void pointer.
+    Usually a b'^?' is a function pointer, but because the encoding doesn't contain the function signature, you need to manually create a CFUNCTYPE with the proper types, and cast this pointer to it.
+    """
+register_preferred_encoding(b'^?', UnknownPointer)
```

Comment:
The issue with a dual-purpose function like that is that if you forget the second argument in a normal call (like `register_encoding(b"foo")`) it silently does nothing (because it only returns the `_register` function) and there's no way to give any sort of warning in that case.